src/rai/rai/utils/configurator.py

```
@@ -0,0 +1,885 @@
+from langchain_openai import ChatOpenAI, OpenAIEmbeddings
+
+# Initialize session state for tracking steps if not exists
+if "current_step" not in st.session_state:
+    st.session_state.current_step = 1
+if "config" not in st.session_state:
+    # Load initial config from TOML file
+    try:
+        with open("config.toml", "rb") as f:
+            st.session_state.config = tomli.load(f)
+    except FileNotFoundError:
+        raise FileNotFoundError("config.toml not found. Please recreate it.")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)